### ISSUE TITLE
Prisma dependencies installed outside of directory being invoked from in split setup

### ISSUE THREAD

**Issue by shayneczyzewski** [prisma/prisma#11770]
### Bug description

We have a project setup as follows:

```
/db
  /migrations
  dev.db-journal
  schema.prisma
/server
  /node_modules
  .gitignore
  index.js
  package-lock.json
  package.json
```

This means the server is not in charge of the db, but rather we can in the future have the same db used by many different web apps, or some other interesting distributed setups.

The server directory has all Prisma dependencies included, and we run all `npx prisma` commands from there. However, when running commands from the server directory, like: `npx prisma migrate dev --schema ../db/schema.prisma` for example, we see that Prisma installs dependences into the db folder:

```
/db
  /migrations
  /node_modules       👈
  dev.db-journal
  package-lock.json   👈
  package.json        👈
  schema.prisma
/server
  /node_modules
  .gitignore
  index.js
  package-lock.json
  package.json
```

Is this expected? It feels odd since we are saying we want the client installed into `output = "../server/node_modules/.prisma/client"` and we are running the command from the server directory, which has all dependencies required.

Another unfortunate byproduct of this is if we change the Prisma version under server, it will _not_ update the versions under db.

### How to reproduce

- Clone: https://github.com/shayneczyzewski/prisma-db-folder-setup-issue as `git clone https://github.com/shayneczyzewski/prisma-db-folder-setup-issue.git`
- `cd` into server directory
- `npm install` deps
- Run: `npx prisma generate --schema ../db/schema.prisma`
- See you now have `npm install` artifacts (node_modules, package*.json) under db now

### Expected behavior

No node_modules directory or package*.json files under db.

### Prisma information

```
datasource db {
  provider = "sqlite"
  url      = "file:./dev.db"
}

generator client {
  provider = "prisma-client-js"
  output = "../server/node_modules/.prisma/client"
}

model Foo {
  id Int @id @default(autoincrement())
  bar String
}
```

### Environment & setup

- OS: macOS Monterey Version 12.2
- Database: SQLite
- Node.js version: v14.16.0

### Prisma Version

```
prisma                  : 3.9.2
@prisma/client          : 3.9.2
Current platform        : darwin
Query Engine (Node-API) : libquery-engine bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash    : bcc2ff906db47790ee902e7bbc76d7ffb1893009
Studio                  : 0.457.0
```

**Comment by janpio** [comment]
Thanks for the information, that is an a bit uncommon (but probably perfectly reasonable!) setup that our current logic might not take into account. We might want to look at where the CLI is executed first (instead of just where the schema file is located) - but someone will have to reproduce and understand first before we can make a call here.

**Comment by shayneczyzewski** [comment]
Great, thanks so much @janpio! Hopefully the example repository I have linked can help with reproduction, and I am here to assist however helpful as someone takes a look into it. I appreciate it.

**Comment by nikkhn** [comment]
@janpio I have a similar structure and experiencing the same issue! What sort of directory structure is typical/would bypass this issue?

**Comment by shayneczyzewski** [comment]
Hi @janpio, I just wanted to circle back on this because we have a user hitting an issue for this (or possibly a related pathing issue). If you have a `package.json` file above the dir you are running an `npx prisma` command in, it will actually install prisma into the parent dir!

To repro:
1) clone https://github.com/shayneczyzewski/prisma-db-folder-setup-issue and switch to `parent-package` branch
2) follow the instructions above
3) you will see that instead of installing into db like the original issue explained, it installs into parent dir now

Would you like me to create a new issue for this perhaps? I feel like it is related to this one, however. Any way I can help your team validate and triage this? It is biting a few of our Wasp users. Thanks!

**Comment by janpio** [comment]
If you think this is not _exactly_ the same problem a new issue would be nice. We are actually focussing on these _right now_, so we will go through all these soon with a comb to try to understand what is going on there. Thanks!

**Comment by shayneczyzewski** [comment]
> If you think this is not _exactly_ the same problem a new issue would be nice. We are actually focussing on these _right now_, so we will go through all these soon with a comb to try to understand what is going on there. Thanks!

Sounds good, thanks @janpio. Since I am not fully sure, I will create a new issue. Appreciate you and the team giving this some eyes soon. 🙏🏻

**Comment by Martinsos** [comment]
Yup, this is really annoying, it happens to us a lot at Wasp and each time looks like some other error, but really it comes down to this.

**Comment by janpio** [comment]
Do you have a reproduction @Martinsos? Without it, we will have trouble to understand what is going on here.

**Comment by shayneczyzewski** [comment]
It should be the same setup as the original issue. Any ETA on triage and fix for this?

**Comment by janpio** [comment]
I can confirm the situation you describe.

---

Probable explanation: 

`prisma generate` tries to detect if `prisma` and `@prisma/client` are available in your project, for which it looks for a `package.json` in the project and above - but it can not traverse to the side it seems. 

As it can not find anything, it just runs `npm install prisma @prisma/client` in what it defines as the `baseDir`:
```
 prisma:generator  baseDir /workspace/prisma-db-folder-setup-issue/db +0ms
```
In my naive reading of the situation that is just wrong - and should be the `server` directory.

Weirdly, after it installed the dependencies again, it then logs this:
```
  prisma:generator  prismaCLIDir /workspace/prisma-db-folder-setup-issue/server/node_modules/prisma +3s
  prisma:generator  prismaClientDir /workspace/prisma-db-folder-setup-issue/server/node_modules/@prisma/client +0ms
```
Those are the _correct_ dependency paths.

**Comment by shayneczyzewski** [comment]
Thanks for confirming @janpio. I agree it is strange it attempts to traverse up searching and then installs locally, but can then reference the expected directory correctly thereafter. Hopefully, it is a small fix that can be rolled out soon. 🤞🏻 Thanks again!

**Comment by ludralph** [comment]
Closing this in favour of [13164](https://github.com/prisma/prisma/issues/13164)
